### PR TITLE
GSYE-269: Set an empty list as a default children in accumulated_trades

### PR DESCRIPTION
**Pull request by cruky**
This is a precautionary measure in a corner case of missing children key of area in accumulated_trades but the same time there are existed children in area configuration tree.

### CODE REVIEW

**Review by fievelk — DISMISSED** [review]

File: gsy_framework/sim_results/cumulative_grid_trades.py [hunk 1]
```diff
@@ -192,11 +192,11 @@ def _update_area_children_in_accumulated_trades_dict(self, accumulated_trades, a
         Returns: None
         """
         current_child_uuids = [
-            c["uuid"] for c in accumulated_trades[area["uuid"]]["children"]
+            c["uuid"] for c in accumulated_trades[area["uuid"]].get("children", [])
```
Comment (fievelk) [inline gsy_framework/sim_results/cumulative_grid_trades.py:195]: Nit:
```suggestion
            child["uuid"] for child in accumulated_trades[area["uuid"]].get("children", [])
```
I know this wasn't part of the PR, but it might be good to fix these situations while we refactor existing code. Anyway, it's just minor.

Comment (hannesdiedrich) [inline gsy_framework/sim_results/cumulative_grid_trades.py:195]: Can you please explain, what you are trying to achieve here? 
If `children=None`, then the for loop will still fail.

Comment (fievelk) [reply gsy_framework/sim_results/cumulative_grid_trades.py:195]: I think that if `children = None`, there would be an exception. I suppose that's why Kamil is trying to replace the `None` with a `[]`.

File: gsy_framework/sim_results/cumulative_grid_trades.py [hunk 2]
```diff
@@ -192,11 +192,11 @@ def _update_area_children_in_accumulated_trades_dict(self, accumulated_trades, a
...
         ]
         for child in area.get("children", []):
             if child["uuid"] not in current_child_uuids:
-                accumulated_trades[area["uuid"]]["children"].append(
+                accumulated_trades[area["uuid"]].setdefault("children", []).append(
```
Comment (hannesdiedrich) [inline gsy_framework/sim_results/cumulative_grid_trades.py:199]: Sorry, but I also need some info here. Why is this needed? For the case when `children` is not part of the `accumulated_trades[area["uuid"]]` ?

File: gsy_framework/sim_results/cumulative_grid_trades.py (hunk 1 above)
Comment (hannesdiedrich) [reply gsy_framework/sim_results/cumulative_grid_trades.py:195]: But this is not the way to replace `None` with `[]`:
```python
di = {"name": "test", "children": None}
print(di.get("children", []))
# None
```

Comment (fievelk) [reply gsy_framework/sim_results/cumulative_grid_trades.py:195]: Sorry, my answer was incomplete/wrong: `.get()` will help in case `children` is not present at all. If `children` is `None`, then you need something like `.get("children") or []`.

Comment (cruky) [reply gsy_framework/sim_results/cumulative_grid_trades.py:195]: I expected that children will be missing or it will be a list.  e.g. this [line](https://github.com/gridsingularity/gsy-framework/blob/8abdd3fc44527cd5aebccb621feeeba6b38ffe6a/gsy_framework/sim_results/cumulative_grid_trades.py#L229) is creating the entries in the accumulated_trades so there will be no None value for children. This [line](https://github.com/gridsingularity/gsy-framework/blob/8abdd3fc44527cd5aebccb621feeeba6b38ffe6a/gsy_framework/sim_results/cumulative_grid_trades.py#L180) is creating a child dict and there is an option for missing children key. Although, you are right, if somehow there will be children=None, then this is not a fix. I can add it.

Comment (cruky) [reply gsy_framework/sim_results/cumulative_grid_trades.py:195]: The main error was:
```
File "/usr/local/lib/python3.8/site-packages/gsy_framework/sim_results/cumulative_grid_trades.py", line 195, in _update_area_children_in_accumulated_trades_dict
c["uuid"] for c in accumulated_trades[area["uuid"]]["children"]
KeyError: 'children'
```
which probably points to the missing `children` key, and not children: None

File: gsy_framework/sim_results/cumulative_grid_trades.py (hunk 2 above)
Comment (cruky) [reply gsy_framework/sim_results/cumulative_grid_trades.py:199]: We had a `KeyError: 'children'` for `accumulated_trades[area["uuid"]]["children"]`.  Which means that the `accumulated_trades` has an area which is missing `children` key. So the append method will raise an error in such case. That's why I create the `children` list and append to it, and if it's already created then it will use existed one.

File: gsy_framework/sim_results/cumulative_grid_trades.py (hunk 1 above)
Comment (hannesdiedrich) [reply gsy_framework/sim_results/cumulative_grid_trades.py:195]: But is this area supposed to have children? I mean assets do not have children per defenition.

File: gsy_framework/sim_results/cumulative_grid_trades.py (hunk 2 above)
Comment (hannesdiedrich) [reply gsy_framework/sim_results/cumulative_grid_trades.py:199]: Here, https://github.com/gridsingularity/gsy-framework/pull/343#discussion_r861098534 also applies.

File: gsy_framework/sim_results/cumulative_grid_trades.py (hunk 1 above)
Comment (cruky) [reply gsy_framework/sim_results/cumulative_grid_trades.py:195]: I didn't get to the root cause so I don't know for sure. What I suspect is somehow the area in the accumulated_trades was overwritten and lost their children. Because the other way, the leaf node (area without children) will never reach the point of `_accumulate_area_trades`.

Comment (cruky) [reply gsy_framework/sim_results/cumulative_grid_trades.py:195]: Is this area supposed to have children? [_accumulate_area_trades](https://github.com/gridsingularity/gsy-framework/blob/8abdd3fc44527cd5aebccb621feeeba6b38ffe6a/gsy_framework/sim_results/cumulative_grid_trades.py#L87) is reached only by Area type, so I think that it is usually should have children.

File: gsy_framework/sim_results/cumulative_grid_trades.py [hunk 3]
```diff
@@ -192,10 +192,12 @@ def _update_area_children_in_accumulated_trades_dict(self, accumulated_trades, a
         Returns: None
         """
         current_child_uuids = [
-            c["uuid"] for c in accumulated_trades[area["uuid"]]["children"]
+            child["uuid"] for child in accumulated_trades[area["uuid"]].get("children", []) or []
```
Comment (Hassan754) [inline gsy_framework/sim_results/cumulative_grid_trades.py:195]: nit, this should be enough to cover the non existing key and the case when it exists and equals to None
```suggestion
            child["uuid"] for child in accumulated_trades[area["uuid"]].get("children") or []
```

**Review by Hassan754 — APPROVED** [review]

File: gsy_framework/sim_results/cumulative_grid_trades.py (hunk 1 above)
Comment (hannesdiedrich) [reply gsy_framework/sim_results/cumulative_grid_trades.py]: I see. Indeed, then this is only a quick fix and probably not the fix of the root cause. I can also give it a try when you can not find it. Please let me know.

**Review by hannesdiedrich — DISMISSED** [review]
LGTM

**Review by Hassan754 — APPROVED** [review]

**Review by spyrostz — APPROVED** [review]
